src/commands/deploy.rs

```diff
@@ -0,0 +1,249 @@
+use anyhow::{Context, Result};
+use bindle::Id;
+use core::panic;
+use hippo_openapi::apis::{
+    account_api::api_account_createtoken_post,
+    app_api::{api_app_post, ApiAppPostError},
+    channel_api::api_channel_post,
+    configuration::{ApiKey, Configuration},
+    revision_api::{api_revision_post, ApiRevisionPostError},
+    Error,
+};
+use hippo_openapi::models::{
+    ChannelRevisionSelectionStrategy, CreateAppCommand, CreateChannelCommand, CreateTokenCommand,
+    RegisterRevisionCommand,
+};
+use reqwest::header;
+use std::path::PathBuf;
+use structopt::{clap::AppSettings, StructOpt};
+
+const APP_CONFIG_FILE_OPT: &str = "APP_CONFIG_FILE";
+const BINDLE_SERVER_URL_OPT: &str = "BINDLE_SERVER_URL";
+const BINDLE_URL_ENV: &str = "BINDLE_URL";
+const HIPPO_SERVER_URL_OPT: &str = "HIPPO_SERVER_URL";
+const HIPPO_URL_ENV: &str = "HIPPO_URL";
+const STAGING_DIR_OPT: &str = "STAGING_DIR";
+const JSON_MIME_TYPE: &str = "application/json";
+
+/// Package and upload Spin artifacts, notifying Hippo
+#[derive(StructOpt, Debug)]
+#[structopt(
+    about = "Deploy a Spin application",
+    global_settings = &[AppSettings::ColoredHelp, AppSettings::ArgRequiredElseHelp]
+)]
+pub struct DeployCommand {
+    /// Path to spin.toml
+    #[structopt(
+        name = APP_CONFIG_FILE_OPT,
+        short = "f",
+        long = "file",
+        default_value = "spin.toml"
+    )]
+    pub app: PathBuf,
+
+    /// URL of bindle server
+    #[structopt(
+        name = BINDLE_SERVER_URL_OPT,
+        long = "bindle-server",
+        env = BINDLE_URL_ENV,
+    )]
+    pub bindle_server_url: String,
+
+    /// URL of hippo server
+    #[structopt(
+        name = HIPPO_SERVER_URL_OPT,
+        long = "hippo-server",
+        env = HIPPO_URL_ENV,
+        default_value = "https://localhost:5309",
```

Comment:
I think we've had part of this conversation in #349, I don't think we want to have a default value for the Hippo URL.
If I'm using a remote Hippo instance (which we _think_ will be the common use case), this will fail because it's trying to connect to `localhost`.

Reply:
Agree. Removed!